src/billable-services/create-edit/add-billable-service.component.tsx

```diff
@@ -423,7 +423,14 @@ const AddBillableService: React.FC<AddBillableServiceProps> = ({
                         id={`paymentMode-${index}`}
                         invalid={!!errors?.payment?.[index]?.paymentMode}
                         invalidText={errors?.payment?.[index]?.paymentMode?.message}
-                        items={paymentModes ?? []}
+                        items={
+                          paymentModes.filter((mode) => {
+                            const selectedUUIDs = fields
+                              .map((f, i) => (i !== index ? f.paymentMode : null))
+                              .filter(Boolean);
+                            return !selectedUUIDs.includes(mode.uuid) || mode.uuid === field.value;
+                          }) ?? []
```

Comment:
```suggestion
                          paymentModes.filter((mode) => {
			                const selectedUUIDs = fields
			                  .map((f, i) => (i !== index ? f.paymentMode : null))
			                  .filter(Boolean);
			              return !selectedUUIDs.includes(mode.uuid) || mode.uuid === currentField.value;
			              })}
```

Recommend extracting this to a helper function so the logic doesn't ge recomputed on every render. Additionally, these suggested changes fix a variable shadowing bug by renaming the Controller's field prop and remove the unnecessary empty array ( ?? []) in line 432 because .filter always returns an array.

We could extract this to a helper that looks something like this:

```tsx
const getAvailablePaymentModes = <T extends { uuid: string }>(
  allModes: T[],
  allFields: PaymentModeForm[],
  currentIndex: number,
  currentValue: string,
): T[] => {
  const selectedUUIDs = allFields
    .map((f, i) => (i !== currentIndex ? f.paymentMode : null))
    .filter(Boolean);

  return allModes.filter((mode) => !selectedUUIDs.includes(mode.uuid) || mode.uuid === currentValue);
};
```